src/main/java/core/basesyntax/strategy/PurchaseOperation.java

```diff
@@ -0,0 +1,19 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.Inventory;
+
+public class PurchaseOperation implements FruitOperationHandler {
+    @Override
+    public void executeOperation(FruitTransaction transaction, Inventory inventory) {
+        int currentBalance = inventory.getInventory().getOrDefault(transaction.getFruit(), 0);
```

Comment:
Be careful with direct access to the inventory map. It's better to use methods that encapsulate the access to internal structures. Consider adding a method in the `Inventory` class to decrease the quantity of fruit and use that method here.